ext/snowcrash/src/MSON.h

```diff
@@ -29,6 +31,36 @@
 
 namespace mson
 {
+    template <typename T, typename>
+    struct just {
+
+        T content;
+
+        just() noexcept(T()) : content{} {}
+
+        explicit just(T&& c) noexcept : content{ std::move(c) } {}
+        explicit just(const T& c) : content{ c } {}
+
+        const T* operator->() const noexcept
+        {
+            return &content;
+        }
+
+        T* operator->() noexcept
+        {
+            return &content;
+        }
+
+        const T& operator*() const noexcept
+        {
+            return content;
+        }
+
+        T& operator*() noexcept
+        {
+            return content;
+        }
+    };
```

Comment:
Both `OneOf` and `Group` sections are typedef'ed to a vector of `Element`, so the only way to disambiguate based on type (e. g. when visiting the variant) is by introducing seperate types (symbols). Above template achieves that.